android/titanium/src/java/org/appcelerator/titanium/view/TiDrawableReference.java

```diff
@@ -157,10 +157,21 @@ public static TiDrawableReference fromBlob(Activity activity, TiBlob blob)
 	 */
 	public static TiDrawableReference fromUrl(KrollProxy proxy, String url)
 	{
+		Activity activity = TiApplication.getAppCurrentActivity();
+		// Attempt to fetch an activity from the given proxy.
+		if (proxy != null) {
+			activity = proxy.getActivity();
+		}
+
 		if (url == null || url.length() == 0 || url.trim().length() == 0) {
-			return new TiDrawableReference(proxy.getActivity(), DrawableReferenceType.NULL);
+			return new TiDrawableReference(activity, DrawableReferenceType.NULL);
+		}
+
+		if (proxy == null) {
```

Comment:
It looks like this doesn't need to use `proxy.resolveUrl()`, it can be simplified to:
```Java
return fromUrl(activity, TiUrl.resolve(TiC.URL_APP_PREFIX, url, null));
```

Reply:
Oops! I should have been more clear, the whole `if (proxy == null) {` conditional scope can be removed so it looks something like this:
```Java
...
if (url == null || url.length() == 0 || url.trim().length() == 0) {
    return new TiDrawableReference(activity, DrawableReferenceType.NULL);
}
return fromUrl(activity, TiUrl.resolve(TiC.URL_APP_PREFIX, url, null));
```